lib/services/semantic-form-management.js

```diff
@@ -137,10 +139,36 @@ export class SemanticFormManagement {
         }
       }
 
-      /**
-       * NOTE: generate and save tailored TTL if is has been configured.
-       */
-      if (configuration.tailored.source) {
+      // SAVE
+      try {
+        await this.updateSemanticForm(uuid, form);
+      } catch (e) {
+        console.warn(`Failed updating sources for semantic-form with ${uuid}`);
+        console.log(e);
+      }
+    }
+
+    if (form.status === SEMANTIC_FORM_STATUS.NEW) {
+      if (!configuration) {
+        // No configuration means multiple sources found. Can be for example :
+        // form.ttl, form.json, tailored extracted data ...
+        // --> We assume only one form.ttl and use it.
+        const ttlSource = form.sources.filter(source => source.physicalURI.endsWith('form.ttl'))[0];
```

Comment:
Some form of ordering, to avoid blindly relying on the the first element is probably an extra safety barrier (albeit perhaps not ideal,  but this needs deeper thinking)